src/Raven.Server/Utils/MicrosoftLogging/MicrosoftLoggingConfiguration.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Raven.Server.NotificationCenter.Notifications;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Sparrow;
+using Sparrow.Json;
+using Sparrow.Logging;
+using Sparrow.Server.Json.Sync;
+
+namespace Raven.Server.Utils.MicrosoftLogging;
+
+public class MicrosoftLoggingConfiguration : IEnumerable<(string Category, LogLevel LogLevel)>
+{
+    private static readonly Logger Logger = LoggingSource.Instance.GetLogger("Server", nameof(MicrosoftLoggingConfiguration));
+    private const string NotificationKey = "microsoft-configuration-logs-error";
+    private const AlertType AlertType = NotificationCenter.Notifications.AlertType.MicrosoftLogsConfigurationLoadError;
+    
+    private readonly string _notificationId = AlertRaised.GetKey(AlertType, NotificationKey);
+
+    private readonly IEnumerable<KeyValuePair<string, SparrowLoggerWrapper>> _loggers;
+    private readonly NotificationCenter.NotificationCenter _notificationCenter;
+    private  readonly ConcurrentDictionary<StringSegment, LogLevel> _configuration = new ConcurrentDictionary<StringSegment, LogLevel>();
+
+    public MicrosoftLoggingConfiguration(IEnumerable<KeyValuePair<string, SparrowLoggerWrapper>> loggers,NotificationCenter.NotificationCenter notificationCenter)
+    {
+        _loggers = loggers;
+        _notificationCenter = notificationCenter;
+    }
+
+    public LogLevel GetLogLevelForCategory(string categoryName)
+    {
+        LogLevel logLevel;
+        for (int i = categoryName.Length; i > 0; i = categoryName.LastIndexOf(".", i - 1, StringComparison.Ordinal))
+        {
+            var segment = new StringSegment(categoryName, 0, i);
+            if (_configuration.TryGetValue(segment, out logLevel))
+                return logLevel;
+        }
+        //Default configuration
+        if (_configuration.TryGetValue(string.Empty, out logLevel))
+            return logLevel;
+
+        return LogLevel.None;
+    }
+    
+    public IEnumerator<(string Category, LogLevel LogLevel)> GetEnumerator()
+    {
+        foreach (var (category, logLevel) in _configuration)
+        {
+            yield return (Category: category.ToString(), LogLevel: logLevel);
+        }
+    }
+
+    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+    
+    public void ReadConfiguration(string configurationPath, JsonOperationContext context, bool shouldThrow, bool reset = true)
+    {
+        FileStream stream;
+        try
+        {
+            stream = File.Open(configurationPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+        }
+        catch (Exception e)
+        {
+            if (e is FileNotFoundException)
+                return ;
```

Comment:
Maybe dismiss the notification here as well?

Reply:
Done